lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -620,18 +619,27 @@ def add_reference(post, authorization, options = {})
         post[:originalReference] = original_reference
       end
 
-      def add_mpi_data_for_network_tokenization_card(post, payment, options)
-        return if options[:skip_mpi_data] == 'Y'
+      def add_network_tokenization_card(post, payment, options)
+        selected_brand = NETWORK_TOKENIZATION_CARD_SOURCE[options[:wallet_type]&.to_s || payment.source.to_s]
+        if selected_brand
+          post[:selectedBrand] = selected_brand
+          post[:additionalData] = {} unless post[:additionalData]
+          post[:additionalData]['paymentdatasource.type'] = selected_brand
+          post[:additionalData]['paymentdatasource.tokenized'] = options[:wallet_type] ? 'false' : 'true' if selected_brand == 'googlepay'
+        end
+
+        return if options[:skip_mpi_data] == 'Y' || options[:wallet_type]
```

Comment:
Is wallet_type a gsf, or where is it coming from? If so is there a chance that a customer would send this with a `NetworkTokenizationCreditCard` and we would skip mpiData in a case where we might not want to?

Reply:
wallet_type get's passed by us for non-tokenized GooglePays which would be a CreditCard in AM. That's why we want to skip it because they would have cryptogram or eci.

Reply:
Awesome! That is perfect, just making sure it couldn't be passed by mistake on regular CCs.